File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
    let answer = 0;
    const TMap = new Map();
    const TSet = new Set();
    
    for(const t of topping) {
        TMap.set(t, (TMap.get(t) || 0) + 1); 
    }
    
    for(const t of topping) {
        TSet.add(t);
        TMap.set(t, TMap.get(t) - 1);
        if(TMap.get(t) === 0) TMap.delete(t);
        
        if(TSet.size === TMap.size) answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter

def solution(topping):
    answer = 0
    tc = Counter(topping)
    ts = set()
    for t in topping:
        ts.add(t)
        tc[t] -= 1
        if tc[t] == 0:
            tc.pop(t)
        if len(tc) == len(ts):
            answer += 1
    return answer
    